packages/elements-core/src/components/TryIt/TryIt.tsx

```diff
@@ -98,20 +98,37 @@ export const TryIt: React.FC<TryItProps> = ({
 
   const [operationAuthValue, setOperationAuthValue] = usePersistedSecuritySchemeWithValues();
 
-  const servers = React.useMemo(
-    () => getServersToDisplay(httpOperation.servers || defaultServers),
-    [httpOperation.servers],
-  );
+  const servers = React.useMemo(() => {
+    const toDisplay = getServersToDisplay(httpOperation.servers || defaultServers);
+
+    if (mockUrl) {
+      toDisplay.push({
+        name: 'Mock Server',
+        url: mockUrl,
+      });
+    }
+
+    return toDisplay;
+  }, [httpOperation.servers, mockUrl]);
   const firstServer = servers[0] || null;
   const [chosenServer, setChosenServer] = useAtom(chosenServerAtom);
+  const isMockingEnabled = mockUrl && chosenServer?.url === mockUrl;
 
   const hasRequiredButEmptyParameters = allParameters.some(
     parameter => parameter.required && !parameterValuesWithDefaults[parameter.name],
   );
 
   React.useEffect(() => {
-    setChosenServer(firstServer);
-  }, [firstServer, setChosenServer]);
+    const currentUrl = chosenServer?.url;
+
+    // simple attempt to preserve / sync up active server if the URLs are the same between re-renders / navigation
```

Comment:
If you have to put in the comment that it's simple, then it is not simple :trollface: